libwhisper/WhisperHost.h

```diff
@@ -50,11 +50,12 @@ class WhisperHost: public HostCapability<WhisperPeer>, public Interface, public
 public:
 	WhisperHost();
 	virtual ~WhisperHost();
-
 	unsigned protocolVersion() const { return 2; }
+	void cleanup();
+	std::map<h256, Envelope> all() const { ReadGuard l(x_messages); return m_messages; }
+	FixedHash<TopicBloomFilterSize> const& bloom() const { return m_bloom; }
```

Comment:
Is this thread-safe?  Can m_bloom be mutated while another thread is reading?
If so, but the method is only used internally and serially, then consider making method protected and setting a friend class. In either case, document thread-safety.